### ISSUE TITLE
[Bug]: Vector embeddings not create in Azure AI search

### ISSUE THREAD

**Issue by FestinaIsm** [run-llama/llama_index#16087]
### Bug Description

I'm creating the 

vector_store = AzureAISearchVectorStore(  
    search_or_index_client=index_client,  
    filterable_metadata_field_keys=metadata_fields,
    index_name=index_name,  
    index_management=IndexManagement.CREATE_IF_NOT_EXISTS,  
    id_field_key="id",  
    chunk_field_key="content",  
    embedding_field_key="embedding",  
    metadata_string_field_key="metadata",
    doc_id_field_key="doc_id",
    embedding_dimensionality=1536,
    language_analyzer="en.lucene",
    vector_algorithm_type="exhaustiveKnn"
)

and initializing the index_client, together with the 

storage_context = StorageContext.from_defaults(vector_store=vector_store)
Settings.llm = llm
Settings.embed_model = embed_model

They all are initiliazed correctly and the index is created. However, the "embedding" field is not present in the index, meaning that the vector embedding is missing alltogether. 

Any ideas? Thank you!

### Version

0.11.10

### Steps to Reproduce

there is no error, it's just that the vector embeddings are not being created

**Comment by non-nil** [comment]
**tl;dr**: the `retrievable` flag for vector fields (i.e. "embedding" in Llama Index) has changed to default `False` in the latest Azure SDK


After some digging, the issue seems to be introduced by changes in the latest Azure SDK schema:
- the embedding field probably is created and stored in Azure Index but simply cannot be retrieved.
- https://github.com/Azure/azure-sdk-for-python/blob/c7f2cc0c12fda2324cafca4b58a740ad2243cc52/sdk/search/azure-search-documents/azure/search/documents/indexes/_generated/models/_models_py3.py#L7343
- 
![image](https://github.com/user-attachments/assets/145fa36c-6df3-4dc6-a161-6a21cb95ada3)

**Comment by non-nil** [comment]
So if one wants to force the embedding field retrievable, we probably have to override the `_create_index` method:

```python
    def _create_index(self, index_name: Optional[str]) -> None:
        ...

        fields = [
            SimpleField(name=self._field_mapping["id"], type="Edm.String", key=True),
            SearchableField(
                name=self._field_mapping["chunk"],
                type="Edm.String",
                analyzer_name=self._language_analyzer,
            ),
            SearchField(
                name=self._field_mapping["embedding"],

                hidden=False, # Force the `SearchField` to be retrievable   <------------------------

                type=SearchFieldDataType.Collection(SearchFieldDataType.Single),
                searchable=True,
                vector_search_dimensions=self._embedding_dimensionality,
                vector_search_profile_name=self._vector_profile_name,
            ),
            SimpleField(name=self._field_mapping["metadata"], type="Edm.String"),
            SimpleField(
                name=self._field_mapping["doc_id"], type="Edm.String", filterable=True
            ),
        ]

        ...
```

**Comment by aayushrajj** [comment]
I have configured the index, indexer and related skillset (with two skills split text and embedding properly). I can see text split working properly as but no vector embedding is being created and that field remains empty.
is there any resolution?
